src/equations/ideal_glm_mhd_multiion_2d.jl

```diff
@@ -0,0 +1,854 @@
+pressure as a function of the state `u` with the signature `electron_pressure(u, equations::IdealGlmMhdMultiIonEquations2D)`.
+By default, the electron pressure is zero.
+
+The argument `initial_c_h` can be used to set the GLM divergence-cleaning speed. Note that 
+`initial_c_h = 0` deactivates the divergence cleaning. The callback [`GlmSpeedCallback`](@ref)
+can be used to adjust the GLM divergence-cleaning speed according to the time-step size.
+
+References:
+- G. Toth, A. Glocer, Y. Ma, D. Najib, Multi-Ion Magnetohydrodynamics 429 (2010). Numerical 
+  Modeling of Space Plasma Flows, 213–218.
+- A. Rueda-Ramírez, A. Sikstel, G. Gassner, An Entropy-Stable Discontinuous Galerkin Discretization
+  of the Ideal Multi-Ion Magnetohydrodynamics System (2024). Journal of Computational Physics.
+  [DOI: 10.1016/j.jcp.2024.113655](https://doi.org/10.1016/j.jcp.2024.113655).
+
+!!! ATTENTION: In case of more than one ion species, these equations should ALWAYS be used
+    with `source_terms_lorentz`.
+"""
+mutable struct IdealGlmMhdMultiIonEquations2D{NVARS, NCOMP, RealT <: Real,
+                                              ElectronPressure} <:
+               AbstractIdealGlmMhdMultiIonEquations{2, NVARS, NCOMP}
+    gammas::SVector{NCOMP, RealT} # Heat capacity ratios
+    charge_to_mass::SVector{NCOMP, RealT} # Charge to mass ratios
+    electron_pressure::ElectronPressure       # Function to compute the electron pressure
+    c_h::RealT                 # GLM cleaning speed
+    function IdealGlmMhdMultiIonEquations2D{NVARS, NCOMP, RealT,
+                                            ElectronPressure}(gammas
+                                                              ::SVector{NCOMP, RealT},
+                                                              charge_to_mass
+                                                              ::SVector{NCOMP, RealT},
+                                                              electron_pressure
+                                                              ::ElectronPressure,
+                                                              c_h::RealT) where
+             {NVARS, NCOMP, RealT <: Real, ElectronPressure}
+        NCOMP >= 1 ||
+            throw(DimensionMismatch("`gammas` and `charge_to_mass` have to be filled with at least one value"))
+
+        new(gammas, charge_to_mass, electron_pressure, c_h)
+    end
+end
+
+function IdealGlmMhdMultiIonEquations2D(; gammas, charge_to_mass,
+                                        electron_pressure = electron_pressure_zero,
+                                        initial_c_h = convert(eltype(gammas), NaN))
+    _gammas = promote(gammas...)
+    _charge_to_mass = promote(charge_to_mass...)
+    RealT = promote_type(eltype(_gammas), eltype(_charge_to_mass))
+
+    NVARS = length(_gammas) * 5 + 4
+    NCOMP = length(_gammas)
+
+    __gammas = SVector(map(RealT, _gammas))
+    __charge_to_mass = SVector(map(RealT, _charge_to_mass))
```

Comment:
```suggestion
    _gammas = promote(gammas...)
    _charge_to_mass = promote(charge_to_mass...)
    RealT = promote_type(eltype(_gammas), eltype(_charge_to_mass))
    __gammas = SVector(map(RealT, _gammas))
    __charge_to_mass = SVector(map(RealT, _charge_to_mass))
    
    NVARS = length(_gammas) * 5 + 4
    NCOMP = length(_gammas)
```

Reply:
I don't understand this suggestion. Isn't this the same code I added?

Reply:
True, but the order is not reflecting the semantics, i.e., the code as is suggests that you need `NVARS` and `NCOMP` to set `__gammas` and `__charge_to_mass`, which is not the case.

Reply:
Yes. It is done like this in other parts of the code. In my opinion, it is clear from the code where the different variables are needed. Anyway, I applied your suggestion in 3411e3cf10fbefcbc26ed856ab70573c3f5a2071.